frontend/js/main.js

```diff
@@ -0,0 +1,2203 @@
+  fetch(apiUrl + '/typemap', {
+    method: 'GET',
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    }
+  })
+  .then(function (res) {
+    console.log(res.ok);
+    if (res.ok) {
+      res.json().then(function (result) {
+        globalDataTypes = result;
+        localStorage.setItem('globalDataTypeList', JSON.stringify(globalDataTypes));
+      });
+    }
+    else {
+      return Promise.reject(res);
+    }
+  }).catch(function (err) {
+    showSnackbar(err, ' redBg');
+  });
+
+  $("#download-schema").click(function () {
+    downloadFilePaths = JSON.parse(localStorage.getItem('downloadFilePaths'));
+    schemaFilePath = downloadFilePaths.Schema;
+    schemaFileName = schemaFilePath.split('/')[schemaFilePath.split('/').length - 1];
+    filePath = './' + schemaFileName;
+    readTextFile(filePath, function (text) {
+      $("<a />", {
+        "download": schemaFileName + ".txt",
+        "href": "data:application/json;charset=utf-8," + encodeURIComponent(text),
+      }).appendTo("body")
+      .click(function () {
+        $(this).remove()
+      })[0].click()
+    })
+  });
+
+  accordion = document.getElementById("accordion");
+
+  // fetching number of tables in source and spanner schema
+  src_table_num = Object.keys(schemaConversionObj.SrcSchema).length;
+  sp_table_num = Object.keys(schemaConversionObj.SpSchema).length;
+
+  expand_button = document.createElement("button")
+  expand_button.setAttribute('id', 'reportExpandButton');
+  expand_button.innerHTML = "Expand All"
+  expand_button.className = "expand"
+  expand_button.addEventListener('click', function () {
+    if ($(this).html() == 'Expand All') {
+      $(this).html('Collapse All');
+      $('.reportCollapse').collapse('show');
+    }
+    else {
+      $(this).html('Expand All');
+      $('.reportCollapse').collapse('hide');
+    }
+  })
+  accordion.appendChild(expand_button);
+
+  // global edit data type button
+  editButton = document.createElement('button');
+  editButton.setAttribute('id', 'editButton');
+  editButton.className = 'expand right-align';
+  editButton.innerHTML = 'Edit Global Data Type';
+  editButton.addEventListener('click', function () {
+    createEditDataTypeTable();
+    $('#globalDataTypeModal').modal();
+  });
+  accordion.appendChild(editButton);
+
+  var reportUl = document.createElement('ul');
+  reportUl.setAttribute('id', 'reportUl');
+  z = 0;
+
+  for (var x = 0; x < src_table_num; x++) {
+    initialPkSeqId[x] = new Array();
+    initialColNameArray[x] = new Array();
+    constraintTabCell[x] = new Array();
+    primaryTabCell[x] = new Array();
+    notPrimary[x] = new Array();
+    notNullFoundFlag[x] = new Array();
+    keyColumnMap[x] = new Array();
+    pkArray[x] = new Array();
+    selectedConstraints[x] = new Array();
+    spPlaceholder[x] = new Array();
+    count_sp[x] = new Array();
+    count_src[x] = new Array();
+    pksSp[x] = new Array();
```

Comment:
Array instantiation can be simplified. Replace with array literals. []

Reply:
Agreed.. Replaced with []